sympy/core/evalf.py

```diff
@@ -187,13 +223,13 @@ def scaled_zero(mag, sign=1):
         raise ValueError('scaled zero expects int or scaled_zero tuple.')
 
 
-def iszero(mpf, scaled=False):
+def iszero(mpf: tUnion[MPF_TUP, SCALED_ZERO_TUP, None], scaled=False) -> Optional[bool]:
     if not scaled:
         return not mpf or not mpf[1] and not mpf[-1]
     return mpf and isinstance(mpf[0], list) and mpf[1] == mpf[-1] == 1
```

Comment:
Why is the return type `Optional` here? It looks like it always returns a bool to me.

Reply:
`iszero(None, True)` returns `None`. This may happen in https://github.com/sympy/sympy/blob/dfd966437a2ac0551c2c55174187f92dd6cba278/sympy/core/evalf.py#L572
Here `re` is returned by `add_terms`
https://github.com/sympy/sympy/blob/dfd966437a2ac0551c2c55174187f92dd6cba278/sympy/core/evalf.py#L549-L550
thus could be `None`

Reply:
Okay, I see. I wonder if that's intentional. I would have written that more explicitly if I wanted to return None.